### PR TITLE
Atomic ledger operations

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by samuel-rufi — DISMISSED** [review]
To me this looks like a nice solution; this is great! I didn't find any issue with this.

**Review by Adam-Gleave — DISMISSED** [review]

File: bee-ledger/src/workers/consensus/worker.rs [hunk 1]
```diff
@@ -28,15 +29,31 @@ use bee_tangle::{ConflictReason, MsTangle, TangleWorker};
...
+    Output(
+        OutputId,
+        oneshot::Sender<(Result<Option<CreatedOutput>, Error>, LedgerIndex)>,
+    ),
+    /// Event to fetch the outputs of an address.
+    Outputs(
+        Address,
+        oneshot::Sender<(Result<Option<Vec<OutputId>>, Error>, LedgerIndex)>,
+    ),
+}
```
Comment (Alex6323) [inline bee-ledger/src/workers/consensus/worker.rs]: Do we want to put this type into its own module? At some point we may want to add an additional `impl` block to this event type containing some convenience methods, and also impl certain traits for it. Wdyt?

Also: probably typo in doc comment: Event**s**.

Comment (Alex6323) [reply bee-ledger/src/workers/consensus/worker.rs]: Two additional suggestions:
From my understanding of the code those events have a `command`-like notion to them. Events usually don't prescribe what the listener has to do upon receiving them. A command however has this notion. So what I would have done here (just mentioning my viewpoint here!) is to call it `ConsensusWorkerCommand` to indicate what the receiver is supposed to do, and also include the instruction as an imperative to the enum variants, so:
`ConfirmMilestone`, `FetchBalance`, `FetchOutput`, `FetchOutputs`.

Comment (thibault-martinez) [reply bee-ledger/src/workers/consensus/worker.rs]: I changed to Command and the new names. Regarding the 1st comment, I think this is consistent with the rest of the codebase like that, I think it's easier to follow if the Event/Command type is together with the worker that consumes them.

Comment (Alex6323) [reply bee-ledger/src/workers/consensus/worker.rs]: alright 👌

**Review by Alex6323 — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

**Review by Adam-Gleave — DISMISSED** [review]

**Pull request merged**